File: RepoTransBench/target_projects/Java/C#/casidiablo_multidex/public_tests/MultiDexPublicTest.cs
using Xunit;

namespace Casidiablo.MultiDex.Tests.Public
{
    public class MultiDexPublicTest
    {
        [Fact]
        public void TestVersionCheckPublic()
        {
            Assert.False(MultiDex.IsVMMultidexCapable("0.9"));
            Assert.False(MultiDex.IsVMMultidexCapable("1.999.9999"));
            Assert.False(MultiDex.IsVMMultidexCapable("2.0.0"));
            Assert.False(MultiDex.IsVMMultidexCapable("2.0.1"));

            Assert.True(MultiDex.IsVMMultidexCapable("2.10"));
            Assert.True(MultiDex.IsVMMultidexCapable("2.1.1"));
            Assert.True(MultiDex.IsVMMultidexCapable("4.0"));
            Assert.True(MultiDex.IsVMMultidexCapable("10.2"));
            Assert.True(MultiDex.IsVMMultidexCapable("2.1.1.5"));
            Assert.True(MultiDex.IsVMMultidexCapable("2.2.12345"));
            Assert.True(MultiDex.IsVMMultidexCapable("05.5.5"));

            Assert.True(MultiDex.IsVMMultidexCapable("002.001.0001"));
            Assert.False(MultiDex.IsVMMultidexCapable("2.0.9999"));
            Assert.False(MultiDex.IsVMMultidexCapable("2.0.0000"));
            Assert.True(MultiDex.IsVMMultidexCapable("3.0.42"));
        }
    }
}

File: RepoTransBench/source_projects/Java/casidiablo_multidex/library/test/src/android/support/multidex/MultiDexPublicTest.java
package android.support.multidex;

import org.junit.Assert;
import org.junit.Test;

/**
 * Public test for {@link MultiDex} class with different VM version data.
 */
public class MultiDexPublicTest {
    @Test
    public void testVersionCheckPublic() {
        // Null and completely malformed
        Assert.assertFalse(MultiDex.isVMMultidexCapable("0.9"));
        Assert.assertFalse(MultiDex.isVMMultidexCapable("1.999.9999"));
        Assert.assertFalse(MultiDex.isVMMultidexCapable("2.0.0"));  // "2.0.0" should be false
        Assert.assertFalse(MultiDex.isVMMultidexCapable("2.0.1"));  // "2.0.x" should be false

        // True for >=2.1.x
        Assert.assertTrue(MultiDex.isVMMultidexCapable("2.10"));    // 2.10 > 2.1
        Assert.assertTrue(MultiDex.isVMMultidexCapable("2.1.1"));
        Assert.assertTrue(MultiDex.isVMMultidexCapable("4.0"));
        Assert.assertTrue(MultiDex.isVMMultidexCapable("10.2"));
        Assert.assertTrue(MultiDex.isVMMultidexCapable("2.1.1.5"));
        Assert.assertTrue(MultiDex.isVMMultidexCapable("2.2.12345"));
        Assert.assertTrue(MultiDex.isVMMultidexCapable("05.5.5"));

        // Edge, padded zeros
        Assert.assertTrue(MultiDex.isVMMultidexCapable("002.001.0001"));
        Assert.assertFalse(MultiDex.isVMMultidexCapable("2.0.9999"));
        Assert.assertFalse(MultiDex.isVMMultidexCapable("2.0.0000"));
        Assert.assertTrue(MultiDex.isVMMultidexCapable("3.0.42"));
    }
}